### PR TITLE
App: zig example - Hello

### PR DESCRIPTION
**Pull request by kassane**
## Summary

Hello zig.

## Impact

none.

## Testing

Arch Linux - simulation

### CODE REVIEW

File: examples/hello_rust/Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ############################################################################
-# apps/examples/hello_rust/Make.defs
+# apps/examples/hello_zig/Make.defs
```
Comment (xiaoxiang781216) [inline examples/hello_rust/Makefile]: why change hello_rust's Makefile

File: examples/hello_zig/Makefile [hunk 2]
```diff
@@ -23,6 +23,7 @@ include $(APPDIR)/Make.defs
 # Hello, Zig! Example
 
 MAINSRC = hello_zig_main.zig
+ZIGFLAGS += -fsingle-threaded
```
Comment (xiaoxiang781216) [inline examples/hello_zig/Makefile:26]: should it be a common flags?

**Comment by xiaoxiang781216** [comment]
@kassane please reorg your patchset too?

File: examples/hello_zig/Makefile (hunk 2 above)
Comment (kassane) [reply examples/hello_zig/Makefile]: Sorry, this flag is only needed when for example trying to make it compatible with rust_app which requires lib to be single-thread.

**Review by pkarashchenko — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**